### DISCUSSION TITLE
golang version of milvus-io/milvus/pkg

### DISCUSSION THREAD

**Discussion by SleepyBearDad** [milvus-io/milvus#27998] [Q&A and General discussion] [answered]
Hi,

I am trying to develop a golang application that requires milvus-io/milvus/pkg.
Currently the latest version is one that only exposes the old internal APIs, so messages are in the old format (milvus-proto/go-api instead of milvus-proto/go-api/v2).

Would it be possible to release a newer version of mlivus/pkg which contain the new message formats?

Shahar

**Comment by yhmo** [comment]
The milvus-proto/go-api is built from this repo: https://github.com/milvus-io/milvus-proto
If you want to use different versions of milvus-proto, you can build it by yourself.

**Reply by SleepyBearDad** [reply]
Hi,

Thanks for the response, I am using the proto/go-api in the repo you mentioned.
The problem is with the milvus-io/milvus/pkg which contains some additional APIs. milvus-cdc also uses this repo for message queue stuff.
Problem is, I can't use the upstream go repo since it is outdated...

Shahar

**Reply by SimFG** [reply, marked as answer]
you can try to get a temp version of the `milvus/pkg` by the specific commit, like:
`go list -m github.com/milvus-io/milvus/pkg@commitID`

**Reply by SleepyBearDad** [reply]
Thanks, will get the package by ID, just note that using @latest gives an outdated package (from Jun 7th)
